src/main/java/frc/robot/subsystems/drive/motors/CIMSteer.java

```diff
@@ -35,17 +37,32 @@ private double nativeToAngle(double nativeUnits) {
   }
 
   public double getAngle() {
+      // return nativeToAngle(getRawPosition());
     return nativeToAngle(getRawPosition()) - this.angleOffset.getDegrees();
   }
 
   public double getVelocity() {
     return nativeToAngle(getRawVelocity());
   }
-
   public void setAngle(double angle) {
     motor.set(ControlMode.Position, angleToNative(angle+angleOffset.getDegrees()));
+    // motor.set(ControlMode.Position, angleToNative(angle));
```

Comment:
i would ask ChatGPT to simplify and comment it better but it's giving me 500s right now

Reply:
Alright, here's what it thinks you could do for your code:
```java
public void setAngleMod360(double naiveDesiredAngle) {
    // Calculate the change in angle from the current angle to the desired angle
    double delta = naiveDesiredAngle - getAngle();

    // Ensure that the change in angle is within the range of [-180, 180] degrees
    double deltaMod360 = delta % 360;
    deltaMod360 = deltaMod360 < 360 ? deltaMod360 + 360 : deltaMod360;
    deltaMod360 = deltaMod360 > 180 ? deltaMod360 - 360 : deltaMod360;

    // Debugging output to SmartDashboard
    if (id == 34) {
        SmartDashboard.putNumber("Delta", delta);
        SmartDashboard.putNumber("Delta % 360", deltaMod360);
    }

    // Calculate the final desired angle by adding the change in angle to the current angle
    double desiredAngle = getAngle() + deltaMod360;

    // Set the angle to the desired angle
    setAngle(desiredAngle);
}
```

Reply:
Here's another AI generated possibly correct (maybe some slight rephrasing) comment you could also add:
```
/**
 * Modifies the current angle to be within the range of [-pi, pi] radians by
 * calculating the change in angle from the current angle to the desired angle,
 * and ensuring that the change is within the range of [-180, 180] degrees.
 * This prevents the angle from "cycling back" when the desired angle is close to
 * the current angle but on the other side of the pi radian boundary.
 */
 ```
 also realized I clicked + on the wrong line